Validation/RPCRecHits/src/RPCRecHitValid.cc

```diff
@@ -355,8 +355,7 @@ void RPCRecHitValid::analyze(const edm::Event &event, const edm::EventSetup &eve
   h_eventCount->Fill(1);
 
   // Get the RPC Geometry
-  edm::ESHandle<RPCGeometry> rpcGeom;
-  eventSetup.get<MuonGeometryRecord>().get(rpcGeom);
+  auto rpcGeom = eventSetup.getHandle(rpcGeomToken_);
```

Comment:
I am not sure why two exact geometries are needed in the class, couldn't we recycle them?

Reply:
no problem to make a member variable for a cache, to keep the pointer to the geometry taken at the bookHistogram(). 

Actually, from the sentence in the [twiki page](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_from_a_Record), I assumed values are taken from a cache, probably I misunderstand.
> Behind the scenes, the EventSetup system only runs the algorithm used to create the data in a Record the first time that data is requested after a Record has changed its IOV. All subsequent requests will get the cached value. 

A fix is made in my local branch and small test is ongoing.

Reply:
EventSetup system indeed caches the products, and the overhead of getting it again for each event is negligible. Getting the products directly from EventSetup on each necessary transition is the best general pattern as developer does not have to assume anything about the IOVs of the concerned EventSetup Records. (imagine taking a pointer at beginRun for a product that has changes at lumi boundaries)